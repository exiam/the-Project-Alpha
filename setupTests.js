const Adapter = require('enzyme-adapter-react-16')

require('enzyme').configure({ adapter: new Adapter() })

require('dotenv').config({ path: '.env' })
