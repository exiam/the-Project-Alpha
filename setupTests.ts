import { configure } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
export function setup() {
  configure({ adapter: new Adapter() })
}
