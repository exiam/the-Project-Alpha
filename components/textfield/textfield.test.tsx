import * as React from 'react'
import TextField, { Props, Color } from './textfield.component'
import { setup } from '../../setupTests'
setup()
import { mount, shallow } from 'enzyme'
import toJSON from 'enzyme-to-json'

const fakeProps: Props = {
  text: 'Hello',
  color: Color.good,
}

describe('Nav component', () => {
  describe('rendering', () => {
    it('should render without error', () => {
      mount(<TextField {...fakeProps} />)
    })

    const wrapper = mount(<TextField {...fakeProps} />)
    it('should render a label input', () => {
      expect(wrapper.exists('label')).toBe(true)
    })
    it('should render a text typed input in the label element', () => {
      expect(wrapper.exists('input[type="text"]')).toBe(true)
    })

    it('should render the label text uppercase', () => {
      expect(wrapper.find('label').text()).toBe('HELLO')
    })
  })
  describe('snapshot', () => {
    it('should match with the snapshot', () => {
      expect(toJSON(shallow(<TextField {...fakeProps} />))).toMatchSnapshot()
    })
  })
})
