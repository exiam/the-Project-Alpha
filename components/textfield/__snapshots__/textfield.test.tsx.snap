// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav component snapshot should match with the snapshot 1`] = `
<textfieldcomponent__TextFieldLabel
  color="#68C281"
>
  HELLO
  <textfieldcomponent__TextFieldInput
    color="#68C281"
    type="text"
  />
</textfieldcomponent__TextFieldLabel>
`;
